### PR TITLE
EIP 665 - Add precompile for Ed25519

### PR DESCRIPTION
**Pull request by oberstet**
See also this PR requesting the same https://github.com/ethereum/EIPs/pull/665

The PR here is more complete and takes up the feedback of @pirapira

### CODE REVIEW

File: EIPS/eip-665.md [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+When a transactions contains data that comes with an Ed25519 signature, that proves that the sender of the Ethereum transaction was also in control of the private key (and the data), and this allows the contract to establish an association between the blockchain and the external system or account, and the external system establish the reverse relation.
+
+For example, a contract might check a Ed25519 signed piece of data submitted to the Ethereum transaction like the current block number. That proves to the contract, that the sender is in possession of both the Ethereum private key and the Ed25518 private key, and hence the contract will accept an association between both. This again can be the root anchor for various powerful applications, as now a potentially crypto holding key owner has proven to be in control of some external off-chain system or account, like e.g. a DNS server, a DNS domain, a cluster node and so on.
+
+## Specification
+
+The proposal adds a new precompiled function with the following signature
+
+```
+ed25519verify(bytes32 m, bytes32 pk, bytes32 s1, bytes32 s2) returns (uint8)
```
Comment (Arachnid) [inline EIPS/eip-665.md]: Existing precompiles do not use function selectors (for example, the RSA precompile). I think it'd be a good idea to remain consistent with that, and specify this in terms of call data.

Comment (oberstet) [reply EIPS/eip-665.md]: Not sure what you mean, could you expand? Note that `m` is supposed to be the message to be signed, and is needed by the actual function body.

Comment (Arachnid) [reply EIPS/eip-665.md]: Take a look at http://eips.ethereum.org/EIPS/eip-196 for instance - the precompile takes call data that is the direct encoding of the parameters, with no function selector (eg, it's not a Solidity-standard multi-function contract). All precompiles so far are like this; it'd be a good idea to retain consistency here.

Comment (oberstet) [reply EIPS/eip-665.md]: Ok, got it. Thanks for pointing me in the right direction! So I believe the changes I pushed should address that.

**Comment by Arachnid** [comment]
This is a courtesy notice to let you know that the format for EIPs has been modified slightly. If you want your draft merged, you will need to make some small changes to how your EIP is formatted:

 - Frontmatter is now contained between lines with only a triple dash ('---')
 - Headers in the frontmatter are now lowercase.

If your PR is editing an existing EIP rather than creating a new one, this has already been done for you, and you need only rebase your PR.

In addition, a continuous build has been setup, which will check your PR against the rules for EIP formatting automatically once you update your PR. This build ensures all required headers are present, as well as performing a number of other checks.

Please rebase your PR against the latest master, and edit your PR to use the above format for frontmatter. For convenience, here's a sample header you can copy and adapt:

```
---
eip: <num>
title: <title>
author: <author>
type: [Standards Track|Informational|Meta]
category: [Core|Networking|Interface|ERC] (for type: Standards Track only)
status: Draft
created: <date>
---
```

**Comment by oberstet** [comment]
@gcolvin Awesome, thx for merging!

As I am new here, may I ask for a tip? What is the best / acceptable way to promote or support this EIP?

- Should I come up with an implementation PR for say `go-ethereum`?
- Should I go to Ethereum forums/IRC and nudge people to support this EIP, "organizing support"?
- Anything else? Did I miss some docs?

Thanks again!

**Comment by Arachnid** [comment]
A good first step would be to ask @souptacular to add this to the agenda for the next All Core Devs, and join the meeting to describe your EIP.

Being able to show a POC implementation on one or more clients wouldn't hurt.

**Comment by oberstet** [comment]
@Arachnid thanks again for your hints! Will follow your advice.

**Pull request merged**